app/controllers/application_controller.rb

```diff
@@ -1,5 +1,11 @@
 class ApplicationController < ActionController::Base
+  before_filter :set_mailer_host
+
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def set_mailer_host
+    ActionMailer::Base.default_url_options[:host] = request.host_with_port
```

Comment:
configuration stuff like like can go in either `config/application.rb` or `config/environments/env-specific-config`.  that way it's set when the app loads up.